dll/win32/shlwapi/ordinal.c

```diff
@@ -3310,10 +3344,24 @@ DWORD WINAPI SHGetIniStringW(LPCWSTR appName, LPCWSTR keyName, LPWSTR out,
         *out = 0;
 
     HeapFree(GetProcessHeap(), 0, buf);
+#endif
 
     return strlenW(out);
 }
 
+#ifdef __REACTOS__
+static BOOL Is7BitClean(LPCWSTR psz)
+{
+    while (*psz)
```

Comment:
The function Is7BitClean() would crash if a nullptr is passed. You use it on only one place in your code (where you check the ptr to be null at the caller) and you do not export the func (which allows you to implement it to suit your needs). The combination of those facts would allow to check for null within, instead of on the outside and make it safe against the crashes. For that you would have to return TRUE on nullptr, meaning "a nullptr is 7-bit-clean".

Reply:
`NULL` check added.